### ISSUE TITLE
[iOS] Bug when Login using GitLab SSO

### ISSUE THREAD

**Issue by mcrainydrop** [mattermost/mattermost-mobile#4712]
#### Summary
Login using GitLab SSO shows error message (see screenshot) in last step of the login procedure, i.e. adding/updating device to list of user devices.
Login procedure according oauth works fine, but this last step fails.

#### Environment Information

- Device Name: iPhone
- OS Version: 13.6.1
- Mattermost App Version: 1.34.0 (Build 318)
- Mattermost Server Version: 5.25.0 (Build 5.25.2)

#### Steps to reproduce
New installation of Mattermost app and select GitLab after entering server url.

#### Expected behavior
Login using GitLab SSO succeeds.

#### Observed behavior
Login is aborted with error message shown in screenshot:
"Received invalid response from the server. URL: https://mattermost.tld.de#/api/v4/users/sessions/device"

![errorMattermostIOS](https://user-images.githubusercontent.com/43291285/90889153-e0e75a00-e3b7-11ea-9b55-6b46ebf29207.jpg)

In the appropriate server logs, this request is handled showing this result: HTTP error 405: Method not allowed
[20/Aug/2020:15:59:28 +0200] [405] [0] "PUT / HTTP/2.0" [TLSv1.3/TLS_AES_256_GCM_SHA384] - [Mozilla/5.0 (iPhone; CPU iPhone OS 13_6_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148]

From my understanding, the anchor/hashtag symbol is simply wrong in the URL and results in the query without the appropriate "/api/v4/users/sessions/device" on server side.

Login using local account works perfectly and in the appropriate requests no anchor symbol is used.

#### Possible fixes
Remove the erroneous "#" from the request URL.

**Comment by amyblais** [comment]
We have been unable to reproduce this internally,
@PokeMMO Are you seeing the same server errors as the original reporter?

Are there any other details that you could help share about your config settings, logs, environment, etc. that could could help us reproduce this?

**Comment by amyblais** [comment]
@mcrainydrop @PokeMMO We have a PR open that may fix the issue: https://github.com/mattermost/mattermost-mobile/pull/4731. Would you be open to testing on the [test builds](https://github.com/mattermost/mattermost-mobile/pull/4731#issuecomment-680970145) if the fix works?

**Comment by enahum** [comment]
Only Android is testable as the ipa file for iOS won;t let you install it on your devices.

**Comment by mcrainydrop** [comment]
Ok, I was too slow in further debugging myself.
I also came to the conclusion, that this issue is related to the Client4 url generation part. I was able to 'workaround' this problem for this very request 'api/v4/users/sessions/device' with rewriting the url for PUT / to PUT /api/v4/users/sessions/device (as it was easy recognizable and rewritable), but then the next request for fetching 'users/me' failed with the same core issue 'mattermost.tld.de#/api/v4/users/me'. Obviously, all user-related api calls are affected, as those are executed after the oauth authorization-flow has succeeded.

This pointed me to the loadMe() method in users.ts and consecutively to the getMe method in client4.ts.
It seemed to me that the issue arise from the this.getUserRoute() -> this.getusersRoute() -> this.getBaseRoute() and there '${this.url}${this.urlVersion}', where the setting of this.url also pointed me to the reworked part in sso.js.

Long story short: 
With the help of a colleague, I've tested the fixed Android version, and everything works smoothly as expected!
This pull request indeed fixes the issue.

Thanks for the pull request and your work!

**Comment by amyblais** [comment]
v1.34.1 dot release fixes this issue and will be available in the app store soon.
